base/compiler/abstractinterpretation.jl

```diff
@@ -1834,35 +1834,38 @@ end
 function abstract_call_unionall(argtypes::Vector{Any})
     if length(argtypes) == 3
         canconst = true
+        a2 = argtypes[2]
         a3 = argtypes[3]
+        nothrow = a2 ⊑ TypeVar && (
+                a3 ⊑ Type ||
+                a3 ⊑ TypeVar)
         if isa(a3, Const)
             body = a3.val
         elseif isType(a3)
             body = a3.parameters[1]
             canconst = false
         else
-            return Any
+            return CallMeta(Any, Effects(EFFECTS_TOTAL; nothrow), NoCallInfo())
```

Comment:
```suggestion
            return CallMeta(Any, EFFECTS_THROWS, NoCallInfo())
```

Reply:
This one I think in theory could be proven nothrow, even if we don't model the rt.